lib/spack/spack/spec.py

```diff
@@ -4861,6 +4862,46 @@ def reconstruct_virtuals_on_edges(spec):
             edge.update_virtuals([vspec])
 
 
+def format_path(spec, format_string):
```

Comment:
This should be a method on the Spec class; that will avoid needing to import `spack.spec` in places we otherwise wouldn't. We frequently have problems implementing new features because of circular imports, we should avoid making it worse. And fundamentally, this is an attribute of the spec -- it's how you format the spec to create a path.

Reply:
It was explicitly requested by @haampie that this not be a method on the `Spec`. I'm fine with it either way, but as far as cyclic imports goes: `Spec` is one of our most fundamental objects and should be import-able anywhere

Reply:
I don't feel too strongly, but I don't think Greg's reason is a very good one.